controllers/activemqartemis_reconciler.go

```diff
@@ -1262,6 +1259,40 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) ProcessResources(customResource
 		return equality.Semantic.DeepEqual(deployed.(*netv1.Ingress).Spec, requested.(*netv1.Ingress).Spec)
 	})
 
+	comparator.Comparator.SetComparator(reflect.TypeOf(routev1.Route{}), func(deployed, requested rtclient.Object) bool {
+		deployedRoute := deployed.(*routev1.Route)
+		requestedRoute := requested.(*routev1.Route)
+
+		reqLogger.V(1).Info("comparing routes", "deployed", *deployedRoute, "requested", *requestedRoute)
+
+		if deployedRoute.Name != requestedRoute.Name {
+			reqLogger.V(1).Info("Route have different name", "deployed", deployedRoute.Name, "requested", requestedRoute.Name)
+			return false
+		}
+
+		if !reflect.DeepEqual(deployedRoute.Spec.Port.TargetPort, requestedRoute.Spec.Port.TargetPort) {
+			reqLogger.V(1).Info("Route target port not euqal", "deplyed", deployedRoute.Spec.Port.TargetPort, "requ", requestedRoute.Spec.Port.TargetPort)
+			return false
+		}
+
+		if deployedRoute.Spec.To.Kind != requestedRoute.Spec.To.Kind {
+			reqLogger.V(1).Info("Route To different - Kind", "deployed", deployedRoute.Spec.To, "requested", requestedRoute.Spec.To)
+			return false
+		}
+
+		if deployedRoute.Spec.To.Name != requestedRoute.Spec.To.Name {
+			reqLogger.V(1).Info("Route To different - Name", "deployed", deployedRoute.Spec.To, "requested", requestedRoute.Spec.To)
+			return false
+		}
+
+		if !reflect.DeepEqual(deployedRoute.Spec.TLS, requestedRoute.Spec.TLS) {
+			reqLogger.V(1).Info("Route tls not equal", "deployed", deployedRoute.Spec.TLS, "requested", requestedRoute.Spec.TLS)
+			return false
+		}
+		reqLogger.V(1).Info("routes are equal")
+		return true
+	})
+
```

Comment:
This is to improve reconcile of route resources. The route has some other fields that are auto filled by the openshift, next time when it reconcile again the "deployed" route will be "different" from the expected route object, causing unnecessary resource updates. (I think the same is true with ingress, which is why we added a comparator for ingress too).

Reply:
Ingress comparator was added because the default comparator doesn't include it while it already includes a comparator for Route, for further details see https://github.com/RHsyseng/operator-utils/blob/v1.4.10/pkg/resource/compare/defaults.go#L98

I see the test `console and acceptor share one secret` passing also without this comparator.

The default Route comparator seems similar to the one added, what does the custom comparator do that the default doesn't?